### PR TITLE
fix(community) : Upgrade node-llama-cpp to be compatible with version 3 

### PR DESCRIPTION
**Pull request by rd4cake**
[Work in Progress]

Draft PR to address the following issue: https://github.com/langchain-ai/langchainjs/issues/6994
Using asynchronous function to load models from node-llama-cpp breaks adherence to the default way of instantiating components ( chatmodels, embeddings, llms, etc ).

Here is an example:

before with node-llama-cpp v2
```typescript
const llamaPath = "/path/to/your/model.gguf"; 

const model = new LlamaCpp({ modelPath: llamaPath });
```

after with node-llama-cpp v3
```typescript
const llamaPath = "/path/to/your/model.gguf"; 

const model = await LlamaCpp.llmInit({ modelPath: llamaPath });
```
Question for the maintainers:
Our implementation changes the way LlamaCpp model is instantiated. Is it this fine moving forward?

### CODE REVIEW

**Comment by Jacky3003** [comment]
@jacoblee93 was wondering if it was possible for you to take a look at what we have so far for our implementation, it would be greatly appreciated.

File: examples/src/embeddings/llama_cpp_basic.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { LlamaCppEmbeddings } from "@langchain/community/embeddings/llama_cpp";
 
 const llamaPath = "/Replace/with/path/to/your/model/gguf-llama2-q4_0.bin";
 
-const embeddings = new LlamaCppEmbeddings({
+const embeddings = await LlamaCppEmbeddings.embeddingsInit({
```
Comment (jacoblee93) [inline examples/src/embeddings/llama_cpp_basic.ts:5]: Would prefer these all to be named `.initialize` for consistency

File: libs/langchain-community/src/chat_models/llama_cpp.ts [hunk 2]
```diff
@@ -87,20 +91,35 @@ export class ChatLlamaCpp extends SimpleChatModel<LlamaCppCallOptions> {
     return "ChatLlamaCpp";
   }
 
-  constructor(inputs: LlamaCppInputs) {
+  private constructor(inputs: LlamaCppInputs) {
```
Comment (jacoblee93) [inline libs/langchain-community/src/chat_models/llama_cpp.ts:94]: Does this make inheritance impossible? Should it be `protected` instead?

Comment (jacoblee93) [reply libs/langchain-community/src/chat_models/llama_cpp.ts:94]: And is there no sync API anymore?

Comment (Jacky3003) [reply libs/langchain-community/src/chat_models/llama_cpp.ts:94]: Unfortunately, there isn't a synchronous API anymore so the current implementation is the best workaround right now. This is both outlined in the [docs](https://node-llama-cpp.withcat.ai/guide/chat-session) and the [source code](https://github.com/withcatai/node-llama-cpp/blob/master/src/bindings/Llama.ts#L314). And yea it does make inheritance impossible, so will change to protected, as long as the person using the library is unable to access the constructor directly, all should be fine.

**Review by jacoblee93 — APPROVED** [review]
Seems to run for me, thank you!

**Comment by jacoblee93** [comment]
CC @nigel-daniels

**Comment by jacoblee93** [comment]
Just one merge to main should be fine! If you keep doing it, it requires CI to run again.

**Pull request merged**